app/views/assignments/_form.html.erb

```diff
@@ -215,10 +216,30 @@
                              form: f } %>
       <% end %>
     </div>
-
      <p id='assignment_file_add_link'>
         <%= add_assignment_file_link t('assignment.add_required_file'), f %>
      </p>
+     <br>
+     <% if @show_checkbox_required_files == true %>
+	   <%= f.check_box :only_required_files,
+	                      checked: @assignment.only_required_files,
+	                      onclick: 'set_onbeforeunload(true);' 
+	   %>
+	   <%= f.label :only_required_files,
+	                 t('required_file_warning'),
+	                 { id: "assignment_only_required_files_label" }
+	   %>
+	 <% else %>
+	   <%= f.check_box :only_required_files,
```

Comment:
This behaviour seems very redundant. I think the best thing to do is have the checkbox appear all the time.